[master] Porting #50261 to master - beacons

garethgreenaway opened the pull request:
### What does this PR do?
Porting #50261 to master

### Tests written?
**[NOTICE] Bug fixes or features added to Salt require tests.**
Please review the [test documentation](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite.
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

s0undt3ch committed:
Update to latest `unittest-xml-reporting`. Fix missing `.fileno` method.

s0undt3ch committed:
Update to latest `unittest-xml-reporting`. Fix missing `.fileno` method.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Merge remote-tracking branch 'upstream/master'

dwoz committed:
Linter fixes

dwoz committed:
Linter fixes

dwoz committed:
Linter fixes

dwoz committed:
Linter fixes

dwoz commented:
@garethgreenaway Test failures seem to be valid.

max-arnold commented:
Could you please also backport the multi-engine PR? https://github.com/saltstack/salt/pull/50059

garethgreenaway commented:
@max-arnold thanks. I'll get that one ported over as well.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Fixing failing tests.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

garethgreenaway committed:
Merge branch 'master' into master_port_50261

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Fixing failed test, using a copy of the default minion opts.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

Akm0d committed:
Refactor the mac group tests

 - Simplify setup and teardown logic with user/group.present/absent
 - Added a function for asserting salt-call with no return code
 - Skip tests if required modules and states aren't available

garethgreenaway committed:
various netapi fixes and tests

garethgreenaway committed:
various netapi fixes and tests

Commit:
update neon reference

Commit:
Update 2016.11.9.rst

Commit:
Update 2016.11.9.rst

Commit:
Update 2017.7.3.rst

Commit:
Update 2018.3.1.rst

Commit:
Update 2018.3.3.rst

Commit:
Update bootstrap-salt.sh

Commit:
Update core.py

Commit:
Update aptpkg.py

Commit:
Update bootstrap-salt.sh

Commit:
Merge branch 'master' into update_neon_reference

Commit:
Update index.rst

Commit:
Update man pages 3000

garethgreenaway committed:
fixes to allow test_ssh to work.  including roster file options and options by pass the host key check.

garethgreenaway committed:
fixes to allow test_ssh to work.  including roster file options and options by pass the host key check.

garethgreenaway committed:
Merge remote-tracking branch 'upstream/master'

garethgreenaway committed:
Merge branch 'master_salt_netapi_fixes' of github.com:garethgreenaway/salt into master_salt_netapi_fixes

twangboy committed:
Fix issue with removed kwarg

garethgreenaway committed:
Merge branch 'master' into master_port_50261

Commit:
more neon/3000 updates

Commit:
Merge branch 'master' of https://github.com/saltstack/salt into update_neon_reference

Commit:
Merge branch 'master' into update_neon_reference

Commit:
Merge branch 'update_neon_reference' of https://github.com/frogunder/salt into update_neon_reference

Commit:
one more neon/3000 update

twangboy committed:
Add tests for the reg module

twangboy committed:
Fix some lint

twangboy committed:
Fix version-added, can't use 3000

dwoz committed:
Merge pull request #55807 from Akm0d/mac_group_tests

Mac group tests

twangboy committed:
Change version-added to versionadded

garethgreenaway committed:
Removing logging import to test if this resolves the test failure.

garethgreenaway committed:
Removing logging import to test if this resolves the test failure.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

garethgreenaway committed:
Importing salt.log.setup so hopefully we have access to log.trace.

garethgreenaway committed:
Importing salt.log.setup so hopefully we have access to log.trace.

dwoz committed:
Merge pull request #55865 from garethgreenaway/master_salt_netapi_fixes

[master] various netapi fixes and tests

Commit:
Update version.py

Commit:
set release versions with environment variables in the docs

dwoz committed:
Merge pull request #55880 from frogunder/man_pages_3000

Update man pages 3000

dwoz committed:
Merge pull request #55875 from frogunder/update_neon_reference

update neon reference

Commit:
add previous release notes

garethgreenaway committed:
Adding some code to make sure log.trace is available.

garethgreenaway committed:
Adding some code to make sure log.trace is available.

dwoz committed:
Merge pull request #55903 from bryceml/make_docs_build_using_environment_variables_for_releases

set release versions with environment variables in the docs

twangboy committed:
Fix conflicts

twangboy committed:
Merge branch 'master' into fix_win_reg

twangboy committed:
Neon to 3000

twangboy committed:
Merge branch 'fix_win_reg' of https://github.com/twangboy/salt into fix_win_reg

dwoz committed:
Merge pull request #55893 from twangboy/fix_win_reg

Remove references to `include_default`

garethgreenaway committed:
Renaming beacon/log.py to beacon/log_beacon.py to avoid conflicts when importing logging during tests.

garethgreenaway committed:
Renaming beacon/log.py to beacon/log_beacon.py to avoid conflicts when importing logging during tests.

garethgreenaway committed:
Removing unneeded testing code.

garethgreenaway committed:
Removing unneeded testing code.

garethgreenaway committed:
Porting #50261 to master.

garethgreenaway committed:
Merge branch 'master_port_50261' of github.com:garethgreenaway/salt into master_port_50261

garethgreenaway committed:
Renaming test_log.py to test_log_beacon.py to fix unit.test_module_names.BadTestModuleNamesTestCase.test_module_name_source_match failure.
